src/git.md

```diff
@@ -73,6 +73,55 @@ the same, with some steps skipped:
  2. Make, stage, and commit your additional changes just like before.
  3. Push those changes to your fork: `git push`.
 
+## Troubleshooting git issues
+
+You don't need to clone `rust-lang/rust` from scratch if it's out of date!
+Even if you think you've messed it up beyond repair, there are ways to fix
+the git state that don't require downloading the whole repository again.
+Here are some common issues you might run into:
+
+### I deleted my fork on GitHub!
+
+This is not a problem from git's perspective. If you run `git remote -v`,
+it will say something like this:
+
+```
+$ git remote -v
+origin	https://github.com//rust-lang/rust (fetch)
+origin	https://github.com//rust-lang/rust (push)
+personal	https://github.com/jyn514/rust (fetch)
+personal	https://github.com/jyn514/rust (push)
+```
+
+If you renamed your fork, you can change the URL like this:
+
+```console
+git remote set-url personal <URL>
+```
+
+where the <URL> is your new fork.
+
+### I see 'Untracked Files: src/stdarch'?
+
+This is left over from the move to the `library/` directory.
+Unfortunately, `git rebase` does not follow renames for submodules, so you
+have to delete the directory yourself:
+
+```console
+rm -r src/stdarch
+```
+
+### I see `<<< HEAD`?
+
+You were probably in the middle of a rebase or merge conflict. See
+[Conflicts](#conflicts) for how to fix the conflict. If you don't care about the changes
+and just want to get a clean copy of the repository back, you can use `git reset`:
+
+```console
+# WARNING: this throws out any local changes you've made! Consider resolving the conflicts instead.
+git reset --hard master
```

Comment:
Wait, why not `git rebase --abort`?

Reply:
Because I don't know how they got into this state. They could be in the middle of a rebase, a merge conflict, or maybe they even committed the `<<< HEAD` annotations without realizing. `reset master` will work no matter what happened (since we can trust @bors not to let `<<< HEAD` annotation to hit the master branch).

Reply:
Also, I don't want to make them responsible for figuring it out how it happened - if they care they can read the conflicts sections, if not they'll want the absolutely minimum number of steps to make it work.